developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,336 @@
 Standard library extension modules
 ==================================
 
-In this section, we could explain how to write a CPython extension with the C language, but the topic can take a complete book.
+In this section, we could explain how to write a CPython extension with the
+C language, but the topic can take a complete book. We will however explain
+how to add a new extension module to the standard library, for instance, a
+module responsible for accelerating some parts of the library.
```

Comment:
The current sentence is "In this section, we could explain how to write a CPython extension with the C language, but the topic can take a complete book." so I assumed that "CPython extension" designated whatever extension you added to Python (like a plugin) whether it's in C or C++, internally (committed to main) or externally (linked against `libpython` and discoverable if you provide the .so/.dll files).

The name of the page is "Standard library extension modules" so I used the terminology "module" to be consistent and to highlight that the components are in a "module" file/folderr placed in `Modules/`. 

I just made a quick search and didn't find a lot of occurrences of "CPython extensions" but a lot of "extension module". So maybe it's better to remove "CPython extension" and just use the term "extension module" (because there are also C++ extension modules).